### ISSUE TITLE
training_utils.weighted_masked_objective wrongly use sample_weight to weight the losses

### ISSUE THREAD

**Issue by mpad** [keras-team/keras#13329]
**System information**  
I'm pointing a line of code in the master of engine/training_utilis.py so I don't think all the system info is relevant...
- Have I written custom code : irrelevant 
- OS Platform and Distribution : irrelevant  
- TensorFlow backendyes :  
- TensorFlow version: irrelevant 
- Keras version:  master
- Python version:  irrelevant

**Describe the current behavior**  

The problem I see is on  this line

 https://github.com/keras-team/keras/blob/9080613dbc6f0840d7544bccc416121f0864a7fd/keras/engine/training_utils.py#L419

from the function `weighted_masked_objective()`. I understand the goal of this code is to return the weighted average of the losses, where weights are user-given through the `sample_weight` option of `model.fit()` function.

Line 417 (`score_array *= weights`) in this module correctly  applies the weights to the scores (losses). 
Line 418 is a non-operation (division by 1) if the weights are non-0.
But then to get the weighted average I expect  to return `K.sum(score_array)/K.sum(weights)` which is different than `K.mean(score_array)`

**Describe the expected behavior**  
The function should return the real weight-averaged losses, not the mean of weights x losses.

**Code to reproduce the issue**  
Hopefully not needed if the diagnostic I have from the 3 lines above is correct. Note I might be overlooking something if the case where the array `score_array` and/or `weights` are more complex than 1D arrays (?).

**Comment by fchollet** [comment]
This is "sum over batch size" reduction. It is different from dividing by the sum of the weights like we were doing before (so it is a breaking change, which will be highlighted in the release notes).

You can achieve the same behavior as before by rescaling your sample weights to sum to 1 for each batch.

Sum over batch size is the standard reduction mechanism in tf.keras. It is the only mechanism that works consistently with distributed training.

**Comment by mpad** [comment]
Thank you for the fast answer ! And sorry to insist but just to follow-up and clarify : 

- It's not only in master but I see it in my local keras 2.2.4 installation.
- Just to make sure, please consider the case where batch_size=2, losses=[1., 0.1] and weights=[1., 10.], so that the 2nd element is much more  "important". The weighted-average loss would be 2/11~0.2, that is consistent with the "importance" of the 2nd element. Now with the current behaviour we would get 2./2=1.0 for the loss, that is practically ignoring the good performance of the 2nd element. 

Am I correct ? Is it  really the expected behaviour ?  If so, I  suggest this is strongly advertised in the documentation. As I can see, the doc for `sample_weight` option here (see the fit() function) : 
https://keras.io/models/sequential/
just mentions "weighting the loss function" which, to me, sounds exactly as "weighted-average" and not at all "sum over batch size"
